### PR TITLE
cd: add generate conda package

### PR DESCRIPTION
**Pull request by floralatin**
Add generate `conda` package

### CODE REVIEW

File: conda/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  license: Proprietary
+  license_family: PROPRIETARY
+  license_file: 
+  summary: "SDK for Hubble API at Jina AI."
+  doc_url: "https://docs.jina.ai/fundamentals/executor/hub/"
+
+extra:
+  recipe-maintainers:
+    - "jina-hubble-sdk"
+  feedstock-name: jina
```
Comment (nomagick) [inline conda/meta.yaml:71]: Is this file supposed to be tracked?

File: scripts/generate-conda-recipe.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        meta_dict = yaml.safe_load(f)
+    
+        meta_dict['package'] = {'name': '<{ name|lower }>', 'version': '<{ version }>'}
+
+        # build noarch package
+        meta_dict['build']['noarch'] = 'python'
+        if(meta_dict['requirements'] and meta_dict['requirements']['host']):
+            meta_dict['requirements']['host'] = ['docker-py' if item == 'docker' else item for item in meta_dict['requirements']['host']]
+        if(meta_dict['requirements'] and meta_dict['requirements']['run']):
+            meta_dict['requirements']['run'] = ['docker-py' if item == 'docker' else item for item in meta_dict['requirements']['run']]
```
Comment (nomagick) [inline scripts/generate-conda-recipe.py:51]: I'm no python expert, but I think this is unclear.

Maybe try the GitHub Copilet trick from Florian. 
https://jinaai.slack.com/archives/C01L8SPGAFP/p1666775786003009

File: scripts/generate-conda-recipe.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    )
+    recipe = recipe.replace('<{', '{{').replace('}>', '}}')
+
+    recipe_header = ''
+    for line in save_lines:
+        recipe_header =  f'{recipe_header}{line}'
+    recipe = recipe_header + recipe
+
+    with open(source_path, 'w+') as fp:
+        fp.write(recipe)
```
Comment (nomagick) [inline scripts/generate-conda-recipe.py:68]: Are we supposed to overwrite the source file as well?

File: scripts/generate-conda-recipe.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    with open(target_path, 'w+') as fp:
+        fp.write(recipe)
+
+generate_meta_yaml("conda/floralatin-hubble-sdk/meta.yaml", 'conda/meta.yaml')
```
Comment (nomagick) [inline scripts/generate-conda-recipe.py:73]: I guess this shouldn't be hard coded or at least the name is not correct

**Review by nomagick — CHANGES_REQUESTED** [review]

File: scripts/generate-conda-recipe.py (hunk 4 above)
Comment (floralatin) [reply scripts/generate-conda-recipe.py]: I see, and i will  use `generate_meta_yaml(sys.argv[1], sys.argv[2])` to replace the `generate_meta_yaml("conda/floralatin-hubble-sdk/meta.yaml", 'conda/meta.yaml')`

File: scripts/generate-conda-recipe.py (hunk 2 above)
Comment (floralatin) [reply scripts/generate-conda-recipe.py]: You are right, I create a function looks like below: to replace it
```
def replace_package_name(requirements: list, old: str, new: str):
    """replace package name"""

    replace_requirements = []
    for package_name in requirements:
        if package_name == old:
            replace_requirements.append(new)
        else:
            replace_requirements.append(package_name)
    return replace_requirements
```

File: conda/meta.yaml (hunk 1 above)
Comment (floralatin) [reply conda/meta.yaml]: mmm, It is dynamically generated and we can delete it.

File: scripts/generate-conda-recipe.py (hunk 3 above)
Comment (floralatin) [reply scripts/generate-conda-recipe.py]: we have removed the `conda/meta.yaml` so we just overwrite the `source file`.

it looks like this:
```
def generate_meta_yaml(meta_path: str):
    """Generate conda meta yaml"""
   ....
   ....

   with open(meta_path, 'w+') as fp:
        fp.write(recipe)
```

File: scripts/generate-conda-recipe.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
+import yaml
+import sys
+
+class RecipeDumper(yaml.SafeDumper):
```
Comment (samsja) [inline scripts/generate-conda-recipe.py:4]: @floralatin ask me for the python review

I am not sure to understand what this dumper do and why it is needed

Comment (floralatin) [reply scripts/generate-conda-recipe.py:4]: In order to add a line break between top-level objects and ignore aliases, so i write it

Comment (samsja) [reply scripts/generate-conda-recipe.py:4]: Okay I am not sure that it is conda best practice ...

Comment (samsja) [reply scripts/generate-conda-recipe.py:4]: Do other project do like this ?

Comment (floralatin) [reply scripts/generate-conda-recipe.py:4]: > Do other project do like this ?
In jina core [create-conda-recipe.py](https://github.com/jina-ai/jina/tree/master/scripts#:~:text=create%2Dconda%2Drecipe.py)

**Pull request closed without merging**